src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/GEOS_LandAssimGridComp.F90

```diff
@@ -1146,10 +1146,51 @@ subroutine Initialize(gc, import, export, clock, rc)
          rc=status                                                              &
          )
     _VERIFY(status)
-    
+
     call ESMF_TimeIntervalSet(LandAssim_DT, s=LandAssimDtStep, rc=status)
     _VERIFY(status)
+    
+    _ASSERT(mod(LandAssimDtStep,model_dtstep)==0, "inconsistent inputs for HEARTBEAT_DT and LANDASSIM_DTSTEP")
+
+    call ESMF_TimeIntervalGet(LandAssim_DT, H = h, m = m, s = s, rc=status)
+    _VERIFY(status)
+
+    ! read reference time for land analysis from resource file (default: LandAssim_DT=3z)
+    ! ref_time is hhmmss format
+    hhmmss = h*10000 + m*100 + s
+    call MAPL_GetResource(                                                      &
+         MAPL,                                                                  &
+         landassim_reftime,                                                     &
+         'LANDASSIM_REFTIME:',                                                  &
+         default= hhmmss,                                                       &
+         rc=status                                                              &
+         )
+    _VERIFY(status)
+
+    ref_seconds = landassim_reftime/10000*3600 +mod(landassim_reftime,10000)/100 *60 + mod(landassim_reftime,100)
```

Comment:
I might not be reading this correctly, but I think ref_seconds is simply the ref_time converted into seconds.  To be safer and clearer, I think we should add parentheses in the "minutes" term, as in:
`+(mod(landassim_reftime,10000)/100) *60
`
But that's not the real issue.  If the above is correct, then a ref_time of 073000 will result in 27,000 seconds.  This interval is not what we're looking for.
I think what we need is the following:
dt_ref = mod( ref_seconds, LandAssimDtStep)
For LandAssimDtStep=10800 (3 hours), dt_ref will then be 5400 seconds for **all** of the following ref_times: 013000, 043000, 073000, ..., 223000.

Reply:
Here ref_time is in hhmmss format and the maximum seconds is LandAssimDtStep